### PR TITLE
Add context menu to TagTile

### PR DESCRIPTION
**Pull request by lukebarnard1**
With two options: View Community and Remove, which
removes the tag from the panel.

### CODE REVIEW

File: src/stores/TagOrderStore.js [hunk 1]
```diff
@@ -165,13 +175,15 @@ class TagOrderStore extends Store {
     _mergeGroupsAndTags() {
         const groupIds = this._state.joinedGroupIds || [];
         const tags = this._state.orderedTagsAccountData || [];
+        const removedTags = this._state.removedTagsAccountData || [];
```
Comment (dbkr) [inline src/stores/TagOrderStore.js]: Feels like we may want to turn this into a set (or even store it on the store as a set) to make the membership testing less O(n^2).

File: src/actions/TagOrderActions.js [hunk 2]
```diff
@@ -56,4 +56,51 @@ TagOrderActions.moveTag = function(matrixClient, tag, destinationIx) {
...
+    const tags = TagOrderStore.getOrderedTags();
+    const removedTags = TagOrderStore.getRemovedTagsAccountData() || [];
+
+    if (removedTags.includes(tag)) {
+        // Return a thunk that doesn't do anything, we don't even need
+        // an asynchronous action here, the tag is already removed.
+        return () => {};
+    }
+
+    removedTags.push(tag);
```
[the remark this answers is not in the document]
Comment (lukebarnard1) [reply src/actions/TagOrderActions.js]: Not really, we need to calculate `removedTags` here for the optimistic update.

[the remark this answers is not in the document]
Comment (dbkr) [reply src/actions/TagOrderActions.js]: Oh I see - ok, so having traced through how all the async actions and actions creators work again, the contents of the store gets updated as soon as the event dispatch gets around the event loop (ie. once the `pending` lands), so the window where you could lose the update is fairly minimal. (Actually I think its the same array object that gets updated each time anyway so possibly it doesn't race at all...)

I guess this is OK then.

**Comment by lukebarnard1** [comment]
The tests are failing and it's a Real Failure! :grinning:

**Pull request merged**